### PR TITLE
Fixes some edible material things being inedible/too edible

### PR DESCRIPTION
**Pull request by ArcaneDefence**
## About The Pull Request

Lets you eat other edible material things besides just walls and floor tiles.
Prevents you from eating qdel'd structures or scraped away turfs by spam clicking on it too

## Why It's Good For The Game

I have an immense desire to consume a ketchup-coated meat toilet, and otherwise I am denied this. 
No longer can you feed yourself excessively by repeatedly eating a singular pizza floor tile, either.
image

## Changelog
:cl:
fix: Things made of edible materials are more consistently edible, and can no longer be eaten multiple times over by spam clicking them. One serving per pizza floor only, please.
/:cl:

### CODE REVIEW

File: code/datums/components/food/edible.dm [hunk 1]
```diff
@@ -75,7 +75,7 @@ Behavior that's still missing from this component that original food items had t
 		if (!item.grind_results)
 			item.grind_results = list() //If this doesn't already exist, add it as an empty list. This is needed for the grinder to accept it.
 
-	else if(isturf(parent))
+	else if(isturf(parent) || isstructure(parent))
 		RegisterSignal(parent, COMSIG_ATOM_ATTACK_HAND, .proc/TryToEatTurf)
```
Comment (Jared-Fogle) [inline code/datums/components/food/edible.dm:79]: `TryToEatTurf` is an outdated name now, should be renamed to something else.

**Comment by Jared-Fogle** [comment]
CC @Qustinnus

**Comment by Jared-Fogle** [comment]
is this really a fix though

**Comment by Qustinnus** [comment]
Yes, partially an over-sight/consistency mistake on my end. I didn't actually know that they were not edible, but they should be. Each edible thing should be able to be eaten, no?

File: code/datums/components/food/edible.dm (hunk 1 above)
Comment (Qustinnus) [reply code/datums/components/food/edible.dm:79]: Also, test this with every one of the custom mat item pls.

File: code/datums/components/food/edible.dm [hunk 2]
```diff
@@ -250,28 +250,33 @@ Behavior that's still missing from this component that original food items had t
 		else
 			this_food.reagents.add_reagent(r_id, amount)
 
+///Makes sure the thing hasn't been destroyed or fully eaten to prevent eating phantom edibles
+/datum/component/edible/proc/IsFoodGone(var/atom/owner, mob/living/feeder)
```
Comment (Jared-Fogle) [inline code/datums/components/food/edible.dm]:
```suggestion
/datum/component/edible/proc/IsFoodGone(atom/owner, mob/living/feeder)
```

File: code/datums/components/food/edible.dm [hunk 3]
```diff
@@ -250,28 +250,33 @@ Behavior that's still missing from this component that original food items had t
...
+	if(QDELETED(owner)|| !(IS_EDIBLE(owner)))
+		return TRUE
+	if(!owner.reagents.total_volume)//Shouldn't be needed but it checks to see if it has anything left in it.
```
Comment (Jared-Fogle) [inline code/datums/components/food/edible.dm:257]: Early return--`if (owner.reagents.total_volume) return FALSE`.

File: code/datums/components/food/edible.dm [hunk 4]
```diff
@@ -250,28 +250,33 @@ Behavior that's still missing from this component that original food items had t
...
+		to_chat(feeder, "<span class='warning'>None of [owner] left, oh no!</span>")
+		if(isturf(owner))
+			var/turf/T = owner
+			T.ScrapeAway(1, CHANGETURF_INHERIT_AIR)
```
Comment (Jared-Fogle) [inline code/datums/components/food/edible.dm:261]:
```suggestion
			var/turf/edible_turf = owner
			edible_turf.ScrapeAway(1, CHANGETURF_INHERIT_AIR)
```

Comment (Jared-Fogle) [reply code/datums/components/food/edible.dm:261]: Wait, why is this here at all? `IsFoodGone` has no implication of being impure.

File: code/datums/components/food/edible.dm [hunk 5]
```diff
@@ -250,28 +250,33 @@ Behavior that's still missing from this component that original food items had t
...
+		else
+			qdel(owner)
```
Comment (Jared-Fogle) [inline code/datums/components/food/edible.dm:263]: Same with this, why is this handled in `IsFoodGone` which doesn't imply that it mutates the world?

**Review by Mothblocks — APPROVED** [review]

**Pull request merged**